### PR TITLE
[exd] join support #5

### PR DESCRIPTION
**Pull request by 0xAX**
These patches include:
- join support
- where support
- select support

### CODE REVIEW

File: lib/exd/api/crud.ex [hunk 1]
```diff
@@ -93,11 +93,19 @@ defmodule Exd.Api.Crud do
     end
   end
 
+  @lists ["full_join", "right_join", "left_join", "join", "load", "select"]
+  defp to_list({key, value}) when key in @lists, do: {key, String.split(value, ",")}
```
Comment (liveforeverx) [inline lib/exd/api/crud.ex]: add `is_binary(value)` - value can be already list

File: lib/exd/api/crud.ex [hunk 2]
```diff
@@ -213,8 +221,22 @@ defmodule Exd.Api.Crud do
...
+    {:done, Enum.map(data, fn(r) -> Enum.map((r |> Tuple.to_list), fn(rec) -> Map.to_list(rec) end) |> :lists.flatten end)}
+  end
+  defp format_select_response([row | data]) when is_map(row) do
+    [row | data]
+  end
+
   defp format_data(data, api, opts) do
-    unless_error(data, api, export_data(data, opts))
+    # 'select' response is not __struct__, so we need to  format in other place
+    case format_select_response(data) do
```
Comment (liveforeverx) [inline lib/exd/api/crud.ex]: What happens, if you doesn't get list? (try get ... id:1) What happens, if database is unreachable, try to stop mariadb and make a query?

File: lib/exd/builder/where.ex [hunk 3]
```diff
@@ -1,33 +1,67 @@
...
+
+  defp build_ast([], _join_models, _fields_with_types, ast) do
+    ast
+  end
+
+  defp build_ast([branch | ast], join_models, fields_with_types, new_ast) do
+    case branch do
+      {op, line, tmp_ast} when op in [:'not', :'or', :'and', :'==', :'>', :'<', :'!=', :'>=', :'<='] ->
+        internal_ast = build_ast(tmp_ast, join_models, fields_with_types, [])
+        new_ast = case is_list(new_ast) do
```
Comment (liveforeverx) [inline lib/exd/builder/where.ex]: Use `List.wrap` instead of whole case

File: lib/exd/escript/main.ex [hunk 4]
```diff
@@ -35,8 +35,10 @@ example of usage:
   #{script} options #{example_app}
   #{script} options #{link}
   #{script} get #{link} id:1
-  #{script} get #{link} where:"id < 10" limit:5 offset:5
+  #{script} get #{link} where:"id < 10 and id > 1" limit:5 offset:5
   #{script} get #{link} where:"id < 10" order_by:"id:desc" limit:5 offset:5
+  #{script} get #{link} join:"my_model, my_model2", where:"my_model.id == 10 or my_model.id == 15"
+  #{script} get #{link} join:"my_model, my_model2", where:"my_model.id == 10", select:"my_model.name,my_model2.name"
```
Comment (liveforeverx) [inline lib/exd/escript/main.ex]: Only question, whould it be work as `select:name` too?

Comment (0xAX) [inline lib/exd/escript/main.ex]: Yes :) Soon it will work

**Pull request merged**